eth/downloader/peer.go

```diff
@@ -57,11 +60,10 @@ type peer struct {
 	blockIdle   int32 // Current block activity state of the peer (idle = 0, active = 1)
 	receiptIdle int32 // Current receipt activity state of the peer (idle = 0, active = 1)
 	stateIdle   int32 // Current node data activity state of the peer (idle = 0, active = 1)
-	rep         int32 // Simple peer reputation
 
-	blockCapacity   int32 // Number of blocks (bodies) allowed to fetch per request
-	receiptCapacity int32 // Number of receipts allowed to fetch per request
-	stateCapacity   int32 // Number of node data pieces allowed to fetch per request
+	blockThroughput   float64 // Number of blocks (bodies) measured to be retrievable per second
```

Comment:
Please avoid floats where possible. Throughput can be modelled as uint - simply reverse the measure to be time per block/receipt/state request. Aside from the problems with floats in general, this would also avoid need for type conversion when working with the target and max time allowance values for each request type.